src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -1074,6 +1074,54 @@ impl ToQemuParams for DeviceVirtconsole {
     }
 }
 
+// RTC represents a qemu Real Time Clock configuration.
+#[derive(Debug)]
+struct Rtc {
+    // Base is the RTC start time.
+    base: String,
+
+    // Clock is the is the RTC clock driver.
+    clock: String,
+
+    // DriftFix is the drift fixing mechanism.
+    driftfix: String,
+}
+
+impl Rtc {
+    fn new(base: &str, clock: &str, driftfix: &str) -> Rtc {
+        Rtc {
+            base: base.to_owned(),
+            clock: clock.to_owned(),
+            driftfix: driftfix.to_owned(),
+        }
+    }
+
+    // valid returns true if the RTC structure is valid and complete.
+    fn valid(&self) -> bool {
+        if self.clock != "host" && self.clock != "rt" && self.clock != "vm" {
+            return false;
+        }
+
+        if self.driftfix != "none" && self.driftfix != "slew" {
+            return false;
+        }
+
+        true
+    }
+}
+
+#[async_trait]
+impl ToQemuParams for Rtc {
+    async fn qemu_params(&self) -> Result<Vec<String>> {
+        let mut params = Vec::new();
+        params.push("rtc".to_owned());
```

Comment:
This line should be dropped.  What this function need to output is along the lines of
`-rtc base=utc,driftfix=slew,clock=host`
but as is, the output would apparently be more like
`-rtc rtc,base=utc,driftfix=slew,clock=host`